dace/libraries/blas/nodes/gemv.py

```diff
@@ -83,56 +84,608 @@ def make_sdfg(node, parent_state, parent_sdfg):
+                                     dace.Memlet(f"{mul_out}[__i0]",
+                                                 wcr="lambda x, y: x + y")
+                                 },
+                                 external_edges=True,
+                                 output_nodes=output_nodes)
+
+        add_program = "__y_out = ({} * __y_in) + __tmp".format(node.beta)
+
+        memlet_idx = "__i"
+
+        # addition map
+        if node.beta != 0:
+            state.add_mapped_tasklet("_Add_", {"__i": "0:{}".format(N)}, {
+                "__y_in": dace.Memlet(f"_y[{memlet_idx}]"),
+                "__tmp": dace.Memlet(f"{mul_out}[__i]"),
             },
-            external_edges=True,
-            output_nodes=output_nodes)
+                                     add_program,
+                                     {"__y_out": dace.Memlet("_y[__i]")},
+                                     external_edges=True,
+                                     input_nodes={mul_out: access_tmp})
 
-        if node.beta != 0:
-            add_program = "__y_out = ({} * __y_in) + __tmp".format(
-                _cast_to_dtype_str(node.beta, dtype_a))
+        return sdfg
 
-            memlet_idx = "__i"
 
-            # addition map
-            state.add_mapped_tasklet(
-                "_Add_", {"__i": "0:{}".format(N)}, {
-                    "__y_in": dace.Memlet.simple("_y", memlet_idx),
-                    "__tmp": dace.Memlet.simple(mul_out, "__i"),
-                },
-                add_program, {"__y_out": dace.Memlet.simple("_y", "__i")},
-                external_edges=True,
-                input_nodes={mul_out: access_tmp})
+@dace.library.expansion
+class ExpandGemvFpgaAccumulate(ExpandTransformation):
+    """
+    This FPGA-oriented expansion iterates over the input matrix A in simple
+    row-major order, with optional tiling in both dimensions, where the tiles
+    are also traversed in simple row-major order. This means that y is only
+    written once, but x is read for every tile in the y-dimension. The
+    implementation requires accumulation on the output, and does NOT assume
+    native accumulation for the given data type. Instead it uses multiple
+    partial sums to ensure that II=1, and only writes the final accumulated
+    value once it has been combined from the partial sums."""
+    # The above corresponds to gemv_v1 in FBLAS
+
+    environments = []
+
+    @staticmethod
+    def expansion(node,
+                  parent_state,
+                  parent_sdfg,
+                  tile_size_x=None,
+                  tile_size_y=None,
+                  num_partial_sums=16):
+        """
+        :param node: Node to expand.
+        :param parent_state: State that the node is in.
+        :param parent_sdfg: SDFG that the node is in.
+        :param tile_size_x: Tile size along the dimension of the vector x, also
+                            referred to as "N" in BLAS terminology. If set to
```

Comment:
According to BLAS, whether `x` has size "N" or "M" depends also on the value of "transA"
(the same applies also to `y`)